test/junit/scala/collection/immutable/LazyListTest.scala

```diff
@@ -1000,4 +1001,32 @@ class LazyListTest {
       assertEquals(s"mkString 3 $i", goal(3,i), precyc(3,i).mkString)
     }
   }
+  @Test
```

Comment:
Ideally there should be a test `def tapEach_properlyLazy(): Unit` right after `def map_properlyLazy(): Unit` which asserts laziness; it can be something like the following:

```scala
   @Test
   def tapEach_properlyLazy(): Unit = {
     val op = lazyListOp(_.tapEach(_ + 1)) // or some other invocation of `tapEach` here
     assertRepeatedlyFullyLazy(op)
     assertLazyTailWhenHeadEvaluated(op)
     assertLazyHeadWhenTailEvaluated(op)
     assertLazyHeadWhenNextHeadEvaluated(op)
   }
```

You can also have this method to check that the behaviour is correct, but there are a lot of `assert<Something>` methods you may find helpful for checking laziness (you can look at the other `<op>_properlyLazy()` methods for examples).